botocore/utils.py

```diff
@@ -632,6 +632,15 @@ def check_dns_name(bucket_name):
     return True
 
 
+def switch_to_s3_sigv2_presigner(signature_version, signing_name, **kwargs):
```

Comment:
By docstring, I meant a full docstring, params, return values etc.

Also, "switching" implies some sort of mutation/side effect.  Here all you're doing is figuring out what the signature version name should be for presigning a request and returning that value.  Might be better to have the name reflect that.